src/v/serde/parquet/encoding.cc

```diff
@@ -90,4 +93,40 @@ iobuf encode_plain(chunked_vector<fixed_byte_array_value> vals) {
     return buf;
 }
 
+iobuf encode_levels(uint8_t max_value, const chunked_vector<uint8_t>& levels) {
+    size_t bit_width = std::bit_width(max_value);
+    iobuf buf;
+    if (bit_width == 0) {
+        bytes len = unsigned_vint::to_bytes(levels.size() << 1U);
```

Comment:
Is `levels` also expected to be empty here? Or is only the reverse guaranteed to be true (that if non-zero then it's non-empty)?

Reply:
No it could be all zeros